### ISSUE TITLE
ADL: Cannot find any AMD adapters...

### ISSUE THREAD

**Issue by mjknieser** [nanopool/ClaymoreZECMiner#18]
Hello All.

New rig with two AMD cards and windows 10. Version 12.6. I am getting the...

"ADL: Cannot find any AMD adapters."

...message.  Seems to happen only after some amount of stopping and re-starting "ZecMiner64.exe".

First tries to stop "ZecMiner64.exe" I do a "Ctrl-C" in the command window. Is this the recommended way to stop "ZecMiner64.exe"?

Then I tried a Windows 10 power-on reset start. This working once; however, not consecutive times.

Then I tried a shutdown (power off) and then power on and life seems good.

Some log snips...

<bad...>
23:51:54:494	18d8	ZEC: Authorized
23:51:54:494	18d8	remove first packet 155
23:51:54:509	18d8	new buf size: 155
23:51:54:509	18d8	parse packet: 118
23:51:54:509	18d8	Pool sets new share target: 0x0001fffe (diff: 32768H)
23:51:54:525	18d8	remove first packet 36
23:51:54:525	18d8	new buf size: 36
23:51:54:525	18d8	parse packet: 35
23:51:54:540	18d8	ZEC: Share accepted!
23:51:54:540	18d8	new buf size: 0
23:51:56:494	974	ADL: Cannot find any AMD adapters
23:51:58:807	18d8	got 295 bytes
23:51:58:807	18d8	buf: {"id":null,"method":"mining.notify","params":["14903","04000000","91ad66825d20142d7481d6fedab0a9ca6006af8c1743e2243a19c7e901000000","458b4b8d058f44b43ec0eab2a281ffe50efdc1b5f19faad1782ea5895e8e44ac","0000000000000000000000000000000000000000000000000000000000000000","471b475a","401f041d",true]}
23:51:58:822	18d8	parse packet: 294
23:51:58:822	18d8	new buf size: 0
23:51:59:541	974	ADL: Cannot find any AMD adapters
23:52:02:604	974	ADL: Cannot find any AMD adapters
23:52:05:620	974	ADL: Cannot find any AMD adapters
23:52:08:651	974	ADL: Cannot find any AMD adapters
23:52:11:683	974	ADL: Cannot find any AMD adapters
23:52:14:730	974	ADL: Cannot find any AMD adapters
23:52:17:746	974	ADL: Cannot find any AMD adapters
23:52:20:778	974	ADL: Cannot find any AMD adapters
23:52:23:043	169c	warning: solutions buf overflow, 115 > 32
23:52:23:840	974	ADL: Cannot find any AMD adapters
23:52:23:872	974	ADL: Cannot find any AMD adapters
23:52:23:872	974	em hbt: 0, fm hbt: 78,
23:52:23:872	974	watchdog - thread 0, hb time 140
23:52:23:887	974	watchdog - thread 1, hb time 453
23:52:23:887	974	watchdog - thread 2, hb time 296
23:52:23:887	974	watchdog - thread 3, hb time -16
23:52:23:903	974	watchdog - thread 4, hb time 484
23:52:23:903	974	watchdog - thread 5, hb time 1171
23:52:23:903	974	watchdog - thread 6, hb time 140
23:52:23:903	974	watchdog - thread 7, hb time 828
23:52:26:966	974	ADL: Cannot find any AMD adapters
23:52:30:013	974	ADL: Cannot find any AMD adapters
23:52:33:060	974	ADL: Cannot find any AMD adapters
23:52:36:076	974	ADL: Cannot find any AMD adapters
23:52:36:311	18d8	got 295 bytes

<good...>
23:58:14:181	a88	ZEC: Authorized
23:58:14:197	a88	remove first packet 155
23:58:14:197	a88	new buf size: 155
23:58:14:197	a88	parse packet: 118
23:58:14:338	a88	Pool sets new share target: 0x0001fffe (diff: 32768H)
23:58:14:338	a88	remove first packet 36
23:58:14:338	a88	new buf size: 36
23:58:14:353	a88	parse packet: 35
23:58:14:353	a88	ZEC: Share accepted!
23:58:14:353	a88	new buf size: 0
23:58:15:463	a88	got 295 bytes
23:58:15:463	a88	buf: {"id":null,"method":"mining.notify","params":["14925","04000000","46aec7395ba8b7366ef649a0b420ed067978e67f7aa182362d8e06d62c49e902","ea61cb147da3b216fac3eb98c76250948b61739993051a72a1307d30da1808da","0000000000000000000000000000000000000000000000000000000000000000","e31c475a","524f0a1d",true]}
23:58:15:463	a88	parse packet: 294
23:58:15:463	a88	new buf size: 0
23:58:16:166	1a98	OC v7, Reset control for GPU 0, close miner right now if you want to use default control from Catalyst
23:58:16:322	1a98	OC v7, Reset control for GPU 1, close miner right now if you want to use default control from Catalyst
23:58:19:198	1624	warning: solutions buf overflow, 152 > 32
23:58:19:541	1a98	OC v7, Reset control for GPU 0, close miner right now if you want to use default control from Catalyst
23:58:19:698	1a98	OC v7, Reset control for GPU 1, close miner right now if you want to use default control from Catalyst
23:58:22:870	1a98	GPU 0 temp = 57, old fan speed = 0, new fan speed = 73
23:58:23:026	1a98	GPU 1 temp = 55, old fan speed = 0, new fan speed = 91
23:58:26:198	1a98	GPU 0 temp = 59, old fan speed = 0, new fan speed = 66
23:58:26:355	1a98	GPU 1 temp = 56, old fan speed = 0, new fan speed = 82
23:58:28:495	308	warning: solutions buf overflow, 354 > 32
23:58:29:058	a88	got 295 bytes

Thoughts?

---
Michael

**Comment by jauyzed** [comment]
+1. 

I am seeing the same error in the logs. I have one amd rx 570 card.

**Comment by raz2000** [comment]
I'm getting the same message on Claymore Dual.

**Comment by mjknieser** [comment]
Yeah...  ...if I need to make any changes to the config, I have to power cycle my rig for the fan speed interface stuff to work.

A couple of random thoughts...

- have a run-time 'q' command to gracefully exit the miner program which, I hope, would solve this issue
- have a run-time 'l' command to re-load the config.txt file

---
Michael

**Comment by raz2000** [comment]
There seem to be some bugs with claymore and amd cards. There is the one you mention here, and this one
https://github.com/nanopool/Claymore-Dual-Miner/issues/109#issuecomment-355527171

Maybe they just want us all to use Nvidia cards :)

Can't find any fixes to either.

**Comment by hrebelo** [comment]
My case was the remote desktop connection.

if I run the claymore with remote desktop connection open, or thru the RDP I get that error

**Comment by onlineque** [comment]
+1  I can see the same error.
If I run the miner from local cmd.exe on win10, it sees everything and it's ok.
If I try to install it as a service  by using nssm (http://nssm.cc) it is constantly giving me this error.
With RDP it's the same, gives the same error.

I have tested other miner (ethminer to be exact) and that one is not impacted, so it seems to me to be a bug, not a "feature" of Windows probably.

Any idea if this can be solved somehow ?

**Comment by b0n0** [comment]
hi, try to disable mb gpu from device manager. still error ?
